src/python/WMCore/MicroService/Unified/MSOutput.py

```diff
@@ -324,9 +320,20 @@ def makeSubscriptions(self, workflow):
 
                 try:
                     # In the message bellow we may want to put the list of datasets too
-                    msg = "Making transfer subscriptions for {}".format(workflow['RequestName'])
-                    self.logger.info(msg)
-                    ddmResultList = self.ddm.makeAggRequests(ddmReqList, aggKey='item')
+                    msg = "Making transfer subscriptions for %s"
+                    self.logger.info(msg, workflow['RequestName'])
+
+                    if ddmReqList:
+                        ddmResultList = self.ddm.makeAggRequests(ddmReqList, aggKey='item')
```

Comment:
I'm pretty sure this question has already been asked in your previous PR, but I don't remember the answer.
What happens if one of the DDM requests fail?

Reply:
It is an atomic call to the the external function `makeAggrequests()` and the aggregation is happening inside that function. If this call fails an exception would be raised. If it succeeds then it will return a list of the statuses of all the transfer requests that have been made - one entry in the result list per aggregated submission [1], if any of them fails the relevant entry in the list will be `None` [2].

Then all the statuses are converted accumulated in a status dictionary [3] and later converted to to `True` `False` values. At the end  if any of the there was even one transfer that have been failed then the Workflow will not be marked as done and will be retried on the next run [4].

[1]
https://github.com/dmwm/WMCore/blob/6c7032b85841a535247026ba70b48686fccd2838/src/python/WMCore/Services/DDM/DDM.py#L327 

[2]
https://github.com/dmwm/WMCore/blob/6c7032b85841a535247026ba70b48686fccd2838/src/python/WMCore/Services/DDM/DDM.py#L334

[3]
https://github.com/dmwm/WMCore/blob/6c7032b85841a535247026ba70b48686fccd2838/src/python/WMCore/MicroService/Unified/MSOutput.py#L339-L345

[4]
https://github.com/dmwm/WMCore/blob/6c7032b85841a535247026ba70b48686fccd2838/src/python/WMCore/MicroService/Unified/MSOutput.py#L347-L352

Reply:
Cool. I'm resolving it then